pkg/validation/validation.go

```diff
@@ -0,0 +1,62 @@
+// Package validation provides default Validator's that can be run with a list
+// of arbitrary objects. The defaults exposed here consist of all Validator's
+// implemented by this validation library.
+//
+// Each default Validator runs an independent set of validation functions on
+// a set of objects. To run all implemented Validator's, use DefaultValidators().
+// The Validator will not be run on objects not of the appropriate type.
+
+package validation
+
+import (
+	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
+	"github.com/operator-framework/api/pkg/validation/internal"
+)
+
+// DefaultPackageManifestValidators returns a package manifest Validator that
+// can be run directly with Apply(objs...). Optionally, any additional
+// Validator's can be added to the returned Validators set.
+func DefaultPackageManifestValidators(vals ...interfaces.Validator) interfaces.Validators {
```

Comment:
nit: This is a little confusing to me. I would expect this to be something like `WithDefaultPackageManifestValidator` or `AddDefaultPackageManifestValidator`. Though those don't really make sense w/ zero arguments, while `Default*` does.

Reply:
I'm thinking we implement something like:

```go
type PackageManifestValidator struct{}

func (f PackageManifestValidator) Validate(objs ...interface{}) (results []errors.ManifestResult) {
    ...
}

func (f PackageManifestValidator) WithValidators(vals ...interfaces.Validator) interfaces.Validators {
	return append(vals, f)
}
```

Then we can do:

```go
var PackageManifestValidators = internal.PackageManifestValidator{}

func foo() {
	results := PackageManifestValidators.WithValidators(myvals...).Apply()
}
```

Reply:
I like it!